ompi/group/group.h

```diff
@@ -469,9 +470,9 @@ static inline int ompi_group_proc_lookup_rank (ompi_group_t* group, ompi_proc_t*
      * Otherwise, wild guess: start from a proportional position
      * compared to comm_world position. */
     v = proc->super.proc_name.vpid;
-    v = (v<np)? v: v*ompi_proc_world_size()/np;
+    v = (v < (opal_vpid_t)np)? v: (v*ompi_proc_world_size())%np;
```

Comment:
This change does not start from a proportional position anymore. The problem I think you were trying to fix (v>np is possible at this point) is resolved on line 475 below when we indeed do a (i+v)%np, which returns v in the valid range. 

Example: with world_size()==10, np==5, whatever the initial value of 'v' >= 5, the result is 0.

Reply:
`(v*np)/ompi_proc_world_size()` should be properly proportional and also always lower than np assuming v < world_size()

Reply:
the initial division by `np` would create buckets of `np` processes instead of distributing the starting point evenly across the process range. That's why the `%` seems more fair to me, once the vp was multiplied by the world size.

Reply:
original code was indeed wrong. See the followup comment with a proposed fix.

Reply:
I don't see how it fixes things, it divides with the world size, which would in general be the largest entity in the universe. That also means that all the vpid of all processes will be smaller than this number, which will result in always starting from the 0 index.

Reply:
If I instantiate the same example I used above `(world_size==10, np==5, vpid==5)` I obtain `v * np / world_size = (5*5)/10= 2` (in integer computation, that why parenthesis order matters), which is roughly proportional to the position of v in world, injected in np. 

Using % will always return 0 if world_size and np are divisors. `v * world_size % np = (5*10) % 5 = 0 (because world_size/np=10%5=0)`

Reply:
When the world size is a multiple of the communicator size using module leads to a bad distribution, but one that is more fair in other cases. Going back to the need for the proportional position, why not simply using the vpid (aka. removing all the tertiary conditional) ?

Reply:
I think that's good, it is simpler.

It's not proportional anymore (so the comment above the block must go), but since proportional is an arbitrary heuristic in the first place the new heuristic will presumably perform better in some cases, worse in others. E.g., splitting world in worldsize/np contiguous sections will be faster, meanwhile splitting by odd/even would be slower.